File: go-client-sdk/pkg/models/operations/getbatchcrmlead.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmLeadQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmLeadResponseBody struct {
	InvalidIds []string      `json:"invalidIds,omitempty"`
	Leads      []shared.Lead `json:"leads,omitempty"`
}

type GetBatchCrmLeadRequest struct {
	QueryParams GetBatchCrmLeadQueryParams
}

type GetBatchCrmLeadResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmLeadResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmlead.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmLeadQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmLeadResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];

  @SpeakeasyMetadata({ data: "json, name=leads", elemType: shared.Lead })
  leads?: shared.Lead[];
}


export class GetBatchCrmLeadRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmLeadQueryParams;
}


export class GetBatchCrmLeadResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmLeadResponseBody;
}
